openslides_backend/i18n/translator.py

```diff
@@ -56,7 +56,7 @@ def parse_language_header(self, lang_header: str) -> list[str]:
             code = code.split("-")[0]
             result.append((q, code))
         # sort by quality value and return only the codes
-        return [t[1] for t in sorted(result)]
+        return [t[1] for t in result]
```

Comment:
```suggestion
        return [t[1] for t in sorted(result, key=lambda tup: tup[0])]
```
The sort did not only sort the languages in alphabetical order: It sorted them by the weight value q first. If that wasn't given or equal to another value, it sorted them in alphabetical order.
I will agree that the latter is probably a bug (in this case caused by the way python sorts tuples), but the former definitely isn't.

The above code _should_ sort the languages by weight first and leave empty values in the same order as before.
If, during testing,  it turns out to wrongly prioritize or wrongly not-prioritize languages with no q-value over those with q-value, I'd suggest trying to set q to `"0.0"` instead of `"1.0"` in line 50.